### PR TITLE
Check brew install paths for socket_vmnet

### PR DESCRIPTION
**Pull request by spowelljr**
**Before:**
If no socket_vmnet flags were provided minikube would check the install path if socket_vmnet was installed from source.

**After:**
If no socket_vmnet flags are provided minikube checks the source, brew amd64, and brew arm64 install paths for socket_vmnet.

**Info:**
socket_vmnet_client relative path: `/opt/socket_vmnet/bin/socket_vmnet_client`
socket_vmnet relative path: `/var/run/socket_vmnet`

source install path prefix: `<none>`
arm64 brew install path prefix: `/opt/homebrew`
amd64 brew install path prefix: `/usr/local`

**Before:**
```
# socket_vmnet installed via brew

minikube start --driver qemu --network socket_vmnet
😄  minikube v1.28.0 on Darwin 13.1 (arm64)
✨  Using the qemu2 driver based on user configuration

🙈  Exiting due to NOT_FOUND_SOCKET_VMNET: 


💡  Suggestion: 

    socket_vmnet was not found on the system, resolve by:
    
    Option 1) Installing socket_vmnet:
    
    https://minikube.sigs.k8s.io/docs/drivers/qemu/#networking
    
    Option 2) Using the user network:
    
    minikube start --driver qemu --network user
```

**After:**
```
# socket_vmnet installed via brew

$ minikube start --driver qemu --network socket_vmnet
😄  minikube v1.28.0 on Darwin 13.1 (arm64)
✨  Using the qemu2 driver based on user configuration
👍  Starting control plane node minikube in cluster minikube
🔥  Creating qemu2 VM (CPUs=2, Memory=4000MB, Disk=20000MB) ...
🐳  Preparing Kubernetes v1.26.1 on Docker 20.10.22 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔗  Configuring bridge CNI (Container Networking Interface) ...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🔎  Verifying Kubernetes components...
🌟  Enabled addons: default-storageclass, storage-provisioner
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
```

### CODE REVIEW

File: cmd/minikube/cmd/start_flags.go [hunk 1]
```diff
@@ -281,8 +281,8 @@ func initNetworkingFlags() {
 	startCmd.Flags().Int(sshSSHPort, defaultSSHPort, "SSH port (ssh driver only)")
 
 	// socket vmnet
-	startCmd.Flags().String(socketVMnetClientPath, "/opt/socket_vmnet/bin/socket_vmnet_client", "Path to the socket vmnet client binary")
-	startCmd.Flags().String(socketVMnetPath, "/var/run/socket_vmnet", "Path to socket vmnet binary")
+	startCmd.Flags().String(socketVMnetClientPath, "", "Path to the socket vmnet client binary")
```
Comment (medyagh) [inline cmd/minikube/cmd/start_flags.go:284]: (lets add QEMU only to the beggining of this two flags, so ppl dont read it when not relevant to them)


```Qemu driver only:....
```

File: pkg/minikube/detect/detect.go [hunk 2]
```diff
@@ -140,12 +140,46 @@ func SocketVMNetInstalled() bool {
 	if runtime.GOOS != "darwin" {
 		return false
 	}
-	_, err := os.Stat(viper.GetString("socket-vmnet-path"))
+	return SocketVMNetPath() != "" && SocketVMNetClientPath() != ""
+}
+
+// SocketVMNetPath returns the path of socket_vmnet
```
Comment (medyagh) [inline pkg/minikube/detect/detect.go:146]: Add Qemu driver only (for code readability)

File: pkg/minikube/detect/detect.go [hunk 3]
```diff
@@ -140,12 +140,46 @@ func SocketVMNetInstalled() bool {
...
+func SocketVMNetPath() string {
+	return checkSocketVMnetInstallLocations("socket-vmnet-path", "/var/run/socket_vmnet")
```
Comment (medyagh) [inline pkg/minikube/detect/detect.go:148]: is this correct ("socket-vmnet-path") ? or meant to be a var ?

File: pkg/minikube/detect/detect.go [hunk 4]
```diff
@@ -140,12 +140,46 @@ func SocketVMNetInstalled() bool {
...
+}
+
+// SocketVMNetClientPath returns the path of socket_vmnet_client
+func SocketVMNetClientPath() string {
+	return checkSocketVMnetInstallLocations("socket-vmnet-client-path", "/opt/socket_vmnet/bin/socket_vmnet_client")
```
Comment (medyagh) [inline pkg/minikube/detect/detect.go:153]: same here is that string supposed to be a var?

File: pkg/minikube/detect/detect.go [hunk 5]
```diff
@@ -140,12 +140,46 @@ func SocketVMNetInstalled() bool {
...
+func checkSocketVMnetInstallLocations(flagName, path string) string {
+	p := viper.GetString(flagName)
+	if p != "" {
+		return p
+	}
+	// source install, arm64 brew install, amd64 brew install
+	prefixes := []string{"", "/opt/homebrew", "/usr/local"}
+	for _, prefix := range prefixes {
+		fullPath := prefix + path
+		if fileExists(fullPath) {
```
Comment (medyagh) [inline pkg/minikube/detect/detect.go:169]: I swear I had seen a builtin golang tool to check if a file exists in the path

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/detect/detect.go (hunk 3 above)
Comment (spowelljr) [reply pkg/minikube/detect/detect.go:148]: This is correct, it's a flag key, we check if the user set their own value, and if they do just use what they provided.

File: pkg/minikube/detect/detect.go (hunk 5 above)
Comment (spowelljr) [reply pkg/minikube/detect/detect.go:169]: Everything I see online just points to `os.Stat` and `os.ErrNotExist`

File: cmd/minikube/cmd/start_flags.go (hunk 1 above)
Comment (spowelljr) [reply cmd/minikube/cmd/start_flags.go:284]: We have `(Docker driver only)` at the end of a lot of our flags, think I should follow that convention to make it consistent

Comment (medyagh) [reply cmd/minikube/cmd/start_flags.go:284]: sure that works, and later we can decide to change the style

File: pkg/minikube/detect/detect.go (hunk 5 above)
Comment (medyagh) [reply pkg/minikube/detect/detect.go:169]: how about "exec.LookPath("...")"

Comment (spowelljr) [reply pkg/minikube/detect/detect.go:169]: Unfortunately, socket_vmnet doesn't get installed into the normal bin folders, so by default neither of the files we use are available on the path so `exec.LookPath` won't work.

File: pkg/minikube/detect/detect.go (hunk 3 above)
Comment (medyagh) [reply pkg/minikube/detect/detect.go:148]: sounds good

**Review by medyagh — CHANGES_REQUESTED** [review]
after adding the comments it looks good to be merged

**Pull request merged**